interactions/client/models/utils.py

```diff
@@ -41,16 +52,24 @@ async def command(ctx):
     :rtype:
     """
 
-    def decorator(coro: Callable[..., Awaitable]) -> Callable[..., Awaitable]:
+    def decorator(coro: Callable[..., Union[Awaitable, Coroutine]]) -> Callable[..., Awaitable]:
         from ..context import ComponentContext
 
         @wraps(coro)
-        async def deferring_func(ctx: Union["CommandContext", "ComponentContext"], *args, **kwargs):
+        async def deferring_func(ctx: Union["CommandContext", "ComponentContext", "Extension"], *args, **kwargs):
             try:
                 loop = get_running_loop()
             except RuntimeError as e:
                 raise RuntimeError("No running event loop detected!") from e
-            task: Task = loop.create_task(coro(ctx, *args, **kwargs))
+
+            if isinstance(args[0], (ComponentContext, CommandContext)):
+                self = ctx
+                ctx = list(args).pop(0)
+
+                task: Task = loop.create_task(coro(self, ctx, *args, **kwargs))
+
+            else:
+                task: Task = loop.create_task(coro(ctx, *args, **kwargs))
```

Comment:
This is unnecessary, they will all get passed as normal

Reply:
It gets the context, since it's needed later to defer. If it doesn't check for it, there will be issues with trying to defer an extension

Reply:
Oh I see now